ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -390,14 +390,17 @@ def _open_key_file(path: str) -> typing.TextIO:
                 None,
             )
         except win32file.error as e:
-            if e.winerror == winerror.ERROR_FILE_NOT_FOUND:
+            if (
+                e.winerror == winerror.ERROR_FILE_NOT_FOUND
+                or e.winerror == winerror.ERROR_PATH_NOT_FOUND
+            ):
                 raise FileNotFoundError(errno.ENOENT, e.strerror, path) from e
             elif (
                 e.winerror == winerror.ERROR_ACCESS_DENIED
                 or e.winerror == winerror.ERROR_SHARING_VIOLATION
             ):
                 raise PermissionError(errno.EACCES, e.strerror, path) from e
-            raise
+            raise WindowsError(errno.ENONET, e.strerror, path) from e
```

Comment:
ENONET does not make any sense. How did you choose this error code? Is it a typo for ENOENT?

https://www.man7.org/linux/man-pages/man3/errno.3.html:

>  ENONET Machine is not on the network.

When I suggested raising WindowsError, I didn't realize it's another alias for OSError. 

> exception OSError(errno, strerror[, filename[, winerror[, filename2]]])
>
> winerror
> Under Windows, this gives you the native Windows error code. The [errno](https://docs.python.org/3/library/exceptions.html#OSError.errno) attribute is then an approximate translation, in POSIX terms, of that native error code.
> 
> Under Windows, if the winerror constructor argument is an integer, the [errno](https://docs.python.org/3/library/exceptions.html#OSError.errno) attribute is determined from the Windows error code, and the errno argument is ignored. On other platforms, the winerror argument is ignored, and the [winerror](https://docs.python.org/3/library/exceptions.html#OSError.winerror) attribute does not exist. 

It turns out that OSError can do all the work for us.

```python
>>> OSError(None, "message", None, winerror.ERROR_FILE_NOT_FOUND)
FileNotFoundError(2, 'message', None, 2)
>>> OSError(None, "message", None, winerror.ERROR_PATH_NOT_FOUND)
FileNotFoundError(2, 'message', None, 3)
>>> OSError(None, "message", None, winerror.ERROR_ACCESS_DENIED)
PermissionError(13, 'message', None, 5)
>>> OSError(None, "message", None, winerror.ERROR_SHARING_VIOLATION)
PermissionError(13, 'message', None, 32)
>>> OSError(None, "message", None, winerror.ERROR_INVALID_FUNCTION)
OSError(22, 'message', None, 1)
```

Reply:
Sure. I have changed all the existing raised errors to OSError with appropriate error codes.

Reply:
I don't think you understand. We don't need any of these cases because `OSError` already handles these. Replace all of the cases with an `OSError`:

```python
        except win32file.error as e:
            raise OSError(None, e.strerror, path, e.winerror) from e
```

Don't replace `e.winerror` with `ERROR_INVALID FUNCTION`. That doesn't make any sense.